iModelCore/Units/PrivateAPI/Formatting/FormattingParsing.h

```diff
@@ -144,6 +144,11 @@ struct FormatParsingSet
     //! Returns error codes when FUS does not match the expression.
     //! The input expression signature code mus be provided by the caller
     BEU::Quantity ComposeColonizedQuantity(Formatting::FormatSpecialCodes cod, FormatCP fusP = nullptr);
+    BEU::Quantity ParseAndProcessTokens(Formatting::FormatSpecialCodes cod, FormatCP fusp, BEU::UnitCP inputUnit);
+    // parsing helper methods
+    BEU::Quantity ParseBearingFormat(FormatProblemCode* probCode, FormatCP fusP, BEU::UnitCP unit);
+    BEU::Quantity ParseRatioFormat(FormatProblemCode* probCode,FormatCP fusP, BEU::UnitCP unit);

```

Comment:
@ColinKerr  I would like to change this API, but what I have in mind is breaking. Check the exported method GetQuantity() on line 160 in this same file.
It's about the FormatProblemCode* parameter.
You can just call format->GetProblem() to obtain the exact same value.
What's even worse, you can also call FormatParsingSet->GetProblemCode() (this same class) to obtain the exact same value for the last call to ParseQuantity.
That's 3 mechanisms to obtain the same piece of information which is probably not used by any consumer other than our own tests.

Also, I recently added Quantity->GetProblemCode() which makes these mechanisms unnecessary. The returned Quantity can just indicate the error. Though, it uses a UnitsProblemCode other than a FormatProblemCode, so it's not the exact same thing. FormatProblemCode has been used as a dumping ground for all sorts of issues, including unit conversion problems, which are unrelated to format.